generators/server/templates/src/main/java/package/web/filter/SpaWebFilter.java.ejs

```diff
@@ -33,7 +33,7 @@ public class SpaWebFilter implements WebFilter {
     @Override
     public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
         String path = exchange.getRequest().getURI().getPath();
-        if (!path.startsWith("/api") && !path.startsWith("/management")<% if (authenticationType === 'oauth2') { %> && !path.startsWith("/login")<% } %>
+        if (!path.startsWith("/api") && !path.startsWith("/management") && !path.startsWith("/test")<% if (authenticationType === 'oauth2') { %> && !path.startsWith("/login")<% } %>
```

Comment:
@murdos I completely agree that this is quite ugly :frowning_face: (but as I wrote, so is the exception in `SecurityConfiguration` :wink:) I thought I ran into problems when doing this (it was in fact my first attempt), because in certain configurations being beneath `/api` some other stuff kicked in that let the tests fail. However a quick test with a reactive and a non-reactive config that I did again right now seemed to work, so I'll see what the tests of all configs say about this change! If it's good, I'll fix up that old commit and remove this exception. (if not we could still change the path conditionally for reactive tests to get rid of this exception, but I thought that was somewhat ugly and complex, too)

Reply:
I do remember now, it was because CSRF would kick in then for `session` and `oauth2`. But it seems to work by adding CSRF test support to `ExceptionTranslatorIT`. So I'll try to go that way because it makes the change of prod code unnecessary :smile: 
Also gonna unify CSRF conditions in tests a little... (`<%_ if (authenticationType == 'session' || authenticationType == 'oauth2' || ...) _%>.with(csrf())` is quite ugly) Would of course be even cooler if single tests wouldn't have to react in any way to this config, but it could be applied in a central point, but that is something for a separate PR :wink:

Reply:
I do remember now, it was because CSRF would kick in then for `session` and `oauth2`. But it seems to work by adding CSRF test support to `ExceptionTranslatorIT`. So I'll try to go that way because it makes the change of prod code unnecessary :smile: 
Also gonna unify CSRF conditions in tests a little... (`<%_ if (authenticationType == 'session' || authenticationType == 'oauth2' || ...) _%>.with(csrf())` is quite ugly) Would of course be even cooler if single tests wouldn't have to react in any way to this config, but it could be applied in a central point, but that is something for a separate PR :wink:

Reply:
@murdos I think I figured it out :smiley: Can you check it again? I fixed up the commit where I had added that ugly hack with `/test` and in my last commit I made the `NoOpMailConfiguration` I had introduced to mock the `MailService` for `AccountResourceIT` transparent, i.e. just put `@Configuration` on it and let it be picked up by Spring if it's in the test classpath. I think this makes sense because there was always some stacktrace from `UserResourceIT` because there was no mocking there and I think in the test scope `MailService` should generally be mocked for all tests (except `MailServiceIT`, but that one creates the `MailService` anyway and is thus not affected). Makes caching of contexts between tests more efficient as well.